diesel/src/connection/transaction_manager.rs

```diff
@@ -200,6 +223,34 @@ where
                 transaction_depth.get() - 1
             )),
         };
+
+        if let Some(msg) = transaction_state.previous_serialization_error.take() {
+            // we can safely ignore the result here
+            // as the only other error than the "rollback error" variants
+            // is failing to load the transaction state, but that's something
+            // we have already done above
```

Comment:
> This "fix" just tracks at connection level if we hit a SerializationError or not and special cases the rollback correspondingly. At least from my point of view that seems like a hacky solution.

Curious what you mean by hacky? Or at least how the same logic could be achieved alternately? Mostly asking because this makes sense to me (basically an early return of sorts), and especially in the context of transactions/isolation levels -- any violation of which I would assume the right "correction" would be to immediately rollback what could be rolled back and then just bail on the rest, returning to the caller an error

Reply:
I think I see what you mean. Yeah on one hand I want to say that most DBs are aware of isolation levels and especially the serializable one; but on the other hand, generic implementations are probably best left generic.

> Maybe I should just generalize that to something that would allow other connections to register "rollback" error variants in a similar manner

I'll have to study the code related to transactions to get a better idea of where this new behavior should live, but do agree with taking this approach

Reply:
The code from commit 553e5b575596a6d9d592bf244fcd149f7fce196d I think indeed makes the error more  generalizable but now I have one question that I feel bad for not asking earlier.

For the code below this line, I see that special handling is required to set the `transaction_state.previous_error_relevant_for_rollback` to `Some`; but otherwise, I'm a bit confused when you write "the result can be safely ignored". You're talking about ignoring the result from the new util method `process_commit_error `, right? If so, is it fair to say that we aren't ignoring the result so much as "overwriting" it to ensure that callers get this specific `kind` of error (as opposed to one of the `CommitErrorOutcome` one)  -- is that about right? If so, curious why we'd need to do this. But mostly just making sure I understand what's going on -- apologies if this is an ill-formed question